File: Linked_List/Reverse_a_linked_list/java/src/MyLinkedList.java
public class MyLinkedList<T> {
    Node<T> head = null;

    public void reverse() {
        Node<T> previous = null;
        Node<T> current = this.head;

        while (current != null) {
            Node<T> next = current.next;
            current.next = previous;
            previous = current;
            current = next;
        }

        this.head = previous;
    }

    public void push(T newData) {
        Node<T> newNode = new Node<>(newData);
        newNode.next = this.head;
        this.head = newNode;
    }

    @Override
    public String toString() {
        Node<T> temp = head;
        StringBuilder builder = new StringBuilder();

        while (temp != null) {
            builder.append(temp.data);
            temp = temp.next;

            if (temp != null) {
                builder.append("--->");
            } else {
                builder.append("--->NULL");
            }
        }

        return builder.toString();
    }
}

File: linkedList/Reverse_a_linked_list/kotlin/src/main/kotlin/MyLinkedList.kt
import java.lang.StringBuilder

class MyLinkedList<T> {
    var head: Node<T>? = null

    fun reverse() {
        var previous: Node<T>? = null
        var current = head
        
        while (current != null) {
            val next = current.next
            current.next = previous
            previous = current
            current = next
        }
        head = previous
    }

    fun push(newData: T) {
        val newNode = Node(newData)
        newNode.next = head
        head = newNode
    }

    override fun toString(): String {
        var temp = head
        val builder = StringBuilder()
        while (temp != null) {
            builder.append(temp.data)
            temp = temp.next
            if (temp != null) {
                builder.append("--->")
            } else {
                builder.append("--->NULL")
            }
        }
        return builder.toString()
    }
}